### PR TITLE
`Plagiarism checks`: Add a new verdict for no plagiarism case

### PR DESCRIPTION
**Pull request by akesfeden**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Instructors may decide that there is no plagiarism in case the the student has a good justification in the reply.

### Description

We have a new verdict `NO_PLAGIARISM`. If the instructor decides like that, there are no consequences for the student at all and the Plagiarism Case Button in the client should not be shown in anymore. Instead, there should be a button at the bottom of the page stating “Resolved plagiarism case” where the student can still navigate to the details and see the conversation.

Also, we added a modal prompt when changing Plagiarism Case from "Confirmed" to "Denied" because this is a destructive operation.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exercise with a potential Plagiarism Case

1. Log in to Artemis
2. Navigate to Course Administration
3. Make two similar submissions with both students.
4. As the instructor, set the due date of the exercise to now and run plagiarism check from Exercise Detail page > Check Plagiarism page.
5. Notify a student about the plagiarism.
6. With the notified student, check if the exercise detail page shows the red `Plagiarism Case` button.
7. As the instructor, set the verdict as "No Plagiarism".
8. With the notified student, check if the exercise detail page does not show the red `Plagiarism Case` button but shows an info button at the bottom of the page (see screenshot) with text "Resolved Plagiarism Case".

Testing the modal to change plagiarism case from "Confirmed" to "Denied":
1. Go to a confirmed plagiarism case at Exercise Detail page > Check Plagiarism page.
2. Click "Deny".
3. You should see a modal, accepting it should change the plagiarism case from "Confirmed" to "Denied".

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| plagiarism-case-instructor-detail-view.component.ts | 79.54% | 95.23% |
| plagiarism-cases.service.ts | 71.42% | 100% |
| plagiarism-case-verdict.component.ts | 85.71% | 92.3% |
| plagiarism-header.component.ts | 80% | 100% |
| course-exercise-details.component.ts | 58.47% | 76.92% |
| PlagiarismCaseResource.java | 75% | 93% |

### Screenshots

image

image

image

### CODE REVIEW

File: src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts [hunk 1]
```diff
@@ -60,6 +60,9 @@ export class PlagiarismCaseInstructorDetailViewComponent implements OnInit, OnDe
      * and saves the point deduction in percent
      */
     savePointDeductionVerdict(): void {
+        if (!this.isStudentNotified()) {
+            return;
+        }
```
Comment (JohannesStoehr) [inline src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts:65]: Should we maybe throw an error or send an issue to Sentry in that case?

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html [hunk 2]
```diff
@@ -13,7 +13,11 @@
             [showResult]="showResults && latestRatedResult != undefined"
         >
         </jhi-exercise-details-student-actions>
-        <a *ngIf="plagiarismCaseId" class="btn btn-danger btn-sm me-2" [routerLink]="['/courses', courseId, 'plagiarism-cases', plagiarismCaseId]">
+        <a
+            *ngIf="plagiarismCaseInfo && plagiarismCaseInfo.verdict !== PlagiarismVerdict.NO_PLAGIARISM"
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html:17]:
```suggestion
            *ngIf="plagiarismCaseInfo?.verdict !== PlagiarismVerdict.NO_PLAGIARISM"
```

File: src/main/webapp/i18n/de/plagiarism.json [hunk 3]
```diff
@@ -63,6 +64,7 @@
                     "plagiarism": "Unterschleid",
                     "warning": "Warnung",
                     "pointDeduction": "Punktabzug",
+                    "noPlagiarism": "Kein Unterschleid",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/plagiarism.json:67]:
```suggestion
                    "noPlagiarism": "Kein Plagiat",
```
If you want to keep the distinction between "Plagiat" and "Unterschleif", you have to fix the spelling (last character)

File: src/main/webapp/i18n/de/plagiarism.json [hunk 4]
```diff
@@ -81,7 +83,8 @@
                 "respondedByStudent": "Studierende:r hat eine Aussage gemacht",
                 "notResponded": "Noch nicht beantwortet",
                 "noCases": "Keine Plagiatsfälle in diesem Kurs",
-                "notifyStudent": "Studierende:r benachrichtigen"
+                "notifyStudent": "Studierende:r benachrichtigen",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/plagiarism.json:86]:
```suggestion
                "notifyStudent": "Studierende:n benachrichtigen",
```

File: src/test/javascript/spec/component/plagiarism/plagiarism-cases-instructor-view.component.spec.ts [hunk 5]
```diff
@@ -71,6 +71,13 @@ describe('Plagiarism Cases Student View Component', () => {
         student: { id: 4, login: 'Student 2' },
         exercise: exercise2,
     } as PlagiarismCase;
+    const plagiarismCase5 = {
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/plagiarism/plagiarism-cases-instructor-view.component.spec.ts:74]: Seams like you don't need this object

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Left some comments

**Review by rhubinger — DISMISSED** [review]
Tested on TS1. The new feature works fine.

File: src/test/javascript/spec/component/plagiarism/plagiarism-cases-instructor-view.component.spec.ts (hunk 5 above)
Comment (akesfeden) [reply src/test/javascript/spec/component/plagiarism/plagiarism-cases-instructor-view.component.spec.ts:74]: Thanks, I included it in the tests.

File: src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html (hunk 2 above)
Comment (akesfeden) [reply src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html:17]: We need to disable the button when `plagiarismCaseInfo` is `undefined` so the suggested version won't cover it unfortunately.

File: src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts (hunk 1 above)
Comment (akesfeden) [reply src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts:65]: Alright, I'll throw an error 👍

Comment (akesfeden) [reply src/main/webapp/app/course/plagiarism-cases/instructor-view/detail-view/plagiarism-case-instructor-detail-view.component.ts]: I added the errors here: https://github.com/ls1intum/Artemis/pull/5543/commits/7c83f80470057208f4521546cb3c39a1e4cf6e9d

File: src/main/webapp/i18n/de/plagiarism.json [hunk 6]
```diff
@@ -4,6 +4,8 @@
             "plagiarismDetection": "Plagiatskontrolle",
             "confirm": "Bestätigen",
             "deny": "Ablehnen",
+            "denyAfterConfirmModalTitle": "Wechsel von Bestätigen zu Ablehnen",
+            "denyAfterConfirmModalText": "Bist du dir sicher, dass du die Entscheidung von \"Bestätigung des Plagiats\" in \"Ablehnung\" ändern möchtest? Dadurch wird der entsprechende Plagiatsfall einschließlich der Kommunikation mit dem/der Schüler/in und des Urteils gelöscht und kann nicht rückgängig gemacht werden.",
```
Comment (bassner) [inline src/main/webapp/i18n/de/plagiarism.json:8]:
```suggestion
            "denyAfterConfirmModalText": "Bist du dir sicher, dass du die Entscheidung von \"Bestätigung des Plagiats\" in \"Ablehnung\" ändern möchtest? Dadurch wird der entsprechende Plagiatsfall einschließlich der Kommunikation mit dem/der Studierenden und des Urteils gelöscht und kann nicht rückgängig gemacht werden.",
```

**Review by bassner — CHANGES_REQUESTED** [review]
Just one thing - I don't think we use the term ``Schüler`` anywhere in our translations :D

File: src/main/webapp/i18n/de/plagiarism.json (hunk 6 above)
Comment (akesfeden) [reply src/main/webapp/i18n/de/plagiarism.json]: Thanks, I applied your suggestion 👍

**Review by bassner — APPROVED** [review]
Tested on Staging, works exactly like described 👍
Code LGTM!

**Review by ge65cer — COMMENTED** [review]
Code changes look good to me. Please add the test coverage to the PR description.

**Comment by akesfeden** [comment]
Thanks for the review @ge65cer, I added test coverages.

**Review by JohannesStoehr — APPROVED** [review]
Checked the code. Thanks for incorporating the feedback

**Review by canberkanar — APPROVED** [review]
Good changes are good 👍🏻

**Pull request merged**